### ISSUE TITLE
Filter encrypted tensor indices based on value

### ISSUE THREAD

**Issue by plotkiny** [facebookresearch/CrypTen#204]
##  Feature

Filter encrypted tensor indices based on value.

`import crypten`
`import torch`
`crypten.init()`

`from crypten.mpc import MPCTensor`

`x =  torch.tensor([1,01,1,1,-1,1])`

`label = MPCTensor(1)`
`x_crypten =  MPCTensor(x)`
`positive = x_crypten == label`
`print(positive.get_plain_text()`  # tensor([1,0,1,1,0,1])

Currently I do not see any implemented methods for retrieving indices in tensor x that are equal to the encrypted label. See below for torch implementation.

## Alternatives
With torch we can perform the following operations.

`label =  1`
`positive = x == label`
`indices = positive.nonzero().squeeze()`
`print(indices)` # tensor([0,2,3,5])

**Comment by knottb** [comment]
The implementation you have above should work in the current version of crypten:

```
import crypten
import torch
crypten.init()

from crypten.mpc import MPCTensor

x = torch.tensor([1, -1, 1, 1, -1, 1])

label = MPCTensor(1)
x_crypten = MPCTensor(x)
positive = x_crypten == label
print(positive.get_plain_text()
```
prints:
```
tensor([1., 1., 1., 1., 0., 1.])
```


Note that the `nonzero` function is not supported in crypten as far as I'm aware `__nonzero__()` should evaluate to a boolean expression (i.e. a branch condition), which is not possible using an encrypted value.
